src/common/utils/http.ts

```diff
@@ -0,0 +1,33 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { app } from "electron";
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export type httpResponse = request.Response;
+
+export function httpGet(url: string, options?: request.CoreOptions): Promise<request.Response> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const requestOptions = Object.assign(
+        {},
+        { url },
+        {
+            headers: {
+                "User-Agent": "readium-desktop",
+                "Accept-Language": `${locale}-${locale.toUpperCase},en-US;q=0.7,en;q=0.5`,
```

Comment:
If `locale` == `fr` or `de`, then this works:
`${locale}-${locale.toUpperCase()}`
=>
`fr-FR`
...but note that there are many other `fr` language codes, such as `fr-CA`, `fr-BE`, `fr-LU`, etc.

However, `locale` == `en` results in `en-EN`, which is incorrect (this would normally be `en-US` or `en-GB`, but there are many others, e.g. `en-CA`, `en-AU`, etc.). 

Also note that the full language code can be lower-case.

References:
https://tools.ietf.org/html/rfc7231#section-5.3.5
https://tools.ietf.org/html/bcp47

Reply:
I know

We can't add a best language code than i18n.locale state in App.

what do you think ?

Reply:
Let's just use `${locale}` and assume the app's configured locale will always be simple language codes like `en`, `de` and `fr`.

Reply:
already done, thanks